firedrake/bcs.py

```diff
@@ -58,11 +58,7 @@ def __init__(self, V, g, sub_domain, method="topological"):
         self.function_arg = g
         self.comm = V.comm
         self._original_arg = self.function_arg
-        if sub_domain == "on_boundary":
-            self.sub_domain = \
-                tuple(map(int, V.mesh().topology.exterior_facets.unique_markers))
-        else:
-            self.sub_domain = sub_domain
+        self.sub_domain = sub_domain
```

Comment:
The markers may not mark the entire boundary, so we deal with it when constructing the nodes.